arangod/Cluster/MaintenanceFeature.cpp

```diff
@@ -747,3 +747,29 @@ arangodb::Result MaintenanceFeature::copyAllErrors(errors_t& errors) const {
 }
 
 
+uint64_t MaintenanceFeature::shardVersion (std::string const& shname) const {
```

Comment:
All three routines become redundant if hash differentiated by setState() on move to terminal state.

Reply:
@neunhoef and i discussed this as needed as of now. we can optimise this after GA.